### ISSUE TITLE
Buffer/Memory containment issues.

### ISSUE THREAD

**Issue by MichaelTiernan** [ArsMasiuk/qvge#80]
Just finished building from git repo clone and getting a number of catastrophic failures all related to memory/buffer overflows or violations.

Running it on kubuntu 19.10.

I'm just starting to try and collect information to help troubleshoot it but I wanted to post a notice about it.

**Comment by ArsMasiuk** [comment]
did you build 0.6 branch? or master?
in 0.6, some memory issues have been already fixed.

**Comment by MichaelTiernan** [comment]
I used master. I'll go back and check.

**Comment by MichaelTiernan** [comment]
Yes. I'd used master. (Also using snap of 0.5.4) Both suffer memory issues.

I just checked out 0.6 and tried to build but got some fatal errors.

Before I clutter the landscape, do you want logs or reports on this would you prefer time to finish addressing the known issues? (i.e. Me just wait.)

**Comment by ArsMasiuk** [comment]
Sorry for too late reply... any data/hints from you are very appreciated, thank you in advance..

**Comment by MichaelTiernan** [comment]
I'll feedback what I can.

**Comment by MichaelTiernan** [comment]
May I suggest two things (I think.)

1) Suggest seeing if there's a way to flag each malloc call.
Here's what I saw in my trace:
~~~
writev(2, [{iov_base="malloc(): invalid size (unsorted"..., iov_len=33}, {iov_base="\n", iov_len=1}], 2malloc(): invalid size (unsorted)
) = 34
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7ff1411aa000
rt_sigprocmask(SIG_UNBLOCK, [ABRT], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1], [], 8) = 0
getpid()                                = 12152
gettid()                                = 12152
tgkill(12152, 12152, SIGABRT)           = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=12152, si_uid=10999} ---
+++ killed by SIGABRT (core dumped) +++
~~~

If there was a way to point out at what point the program is when this happens, it might make it easier to sort out.

2) Include a 'debug &&|| verbose' flag in the invocation.

If we can eliminate sections of operations, it might narrow it down.

**Comment by ArsMasiuk** [comment]
Thank you for the feedback.
Did you built the app with OGDF support as well?
And what should one do in order to reproduce the crash? I.e. load some kind of graph etc.?

**Comment by MichaelTiernan** [comment]
This specific instance was using the distributed snap.
I'm going to build the app from the master branch and from the 0.6 branch (flags to set for the builds more than welcome) to then see what I'll get.

One of the places that it craps itself constantly is when, if the gui comes up, if I roll over the second menu item, that crashes quickly.

**Comment by MichaelTiernan** [comment]
I just built from the 0.6.x branch and ran it.
The gui comes up and when I try to open one of the examples I get:

~~~
./bin/qvgeapp 
*** buffer overflow detected ***: ./bin/qvgeapp terminated
Aborted (core dumped)
~~~

I have a log of the build and crash if it helps.

I expect to try to look at the build log a little later and see if there's any warning signs that should have told me anything.

**Comment by ArsMasiuk** [comment]
Hi,
are you expecting the same issues with the latest (0.5.5) version too?

**Comment by MichaelTiernan** [comment]
I'l give it a try in a second.

**Comment by ArsMasiuk** [comment]
hopefully fixed for 0.6

**Comment by MichaelTiernan** [comment]
I know I didn't help enough. I've tried building it and each attempt fails with a buffer error.
Until I know better, I'm going to say that it's not entirely on "your" side of the equasion.
I hope to get some more time to work on soon.
